packages/react/src/index.test.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Important: This test also serves as a point to
+ * import the entire lib for coverage reporting
+ */
+import { expect } from 'chai';
+import { describe, it } from 'vitest';
+import { isJSDOM } from '#test-utils';
+import * as BaseUI from './index';
+
+describe('@base-ui-components/react', () => {
+  it('should have exports', () => {
+    expect(typeof BaseUI).to.equal('object');
+  });
+
+  it('should not have undefined exports', () => {
+    (Object.keys(BaseUI) as (keyof typeof BaseUI)[]).forEach((exportKey) => {
+      expect(Boolean(BaseUI[exportKey])).to.equal(true);
+    });
+  });
+
+  it.skipIf(!isJSDOM)('should have the correct root exports', async () => {
+    const packageJson = await import('../package.json');
+    const subpathExports = packageJson.exports;
+
+    await Promise.all(
+      Object.keys(subpathExports)
+        .filter((key) => !['.', './utils'].includes(key) && !key.startsWith('./unstable-'))
+        .map(async (subpath) => {
+          const importSpecifier = `@base-ui-components/react/${subpath.replace('./', '')}`;
+          const module = await import(importSpecifier);
+
+          Object.keys(module).forEach((exportKey) => {
+            expect((BaseUI as Record<string, unknown>)[exportKey]).not.to.equal(
+              undefined,
+              `${exportKey} (from ${importSpecifier}) was not found in root exports`,
+            );
+          });
+        }),
+    );
+
+    // Ensure the utils are also exported
+    expect(BaseUI.utils).not.to.equal(undefined);
+  });
```

Comment:
In our releases/npm package I see all the utils are published. Should any utils even be exported to the consumer? I thought they were intended for private/internal usage

Reply:
We need a subset of utils published as they are required to build applications: `useForkRef`, `useId`, maybe `useControlled`.

Reply:
Maybe we can have a list of utils we expect to be exported here and fail if there are more or less exported, that way when we add a new util, we will need to make decision of whether it should be public as the test would automatically fail.

Reply:
It might be easier to manage if we had internal utils in a separate directory. This way we'll be able to treat all contents of ./utils as public (which will make us more careful about introducing breaking changes).

Reply:
Anyway, this is outside the scope of this PR. We can discuss it separately.